File: app/src/main/java/me/devsaki/hentoid/fragments/reader/ReaderBrowseModeDialogFragment.java
package me.devsaki.hentoid.fragments.reader;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;

import me.devsaki.hentoid.R;
import me.devsaki.hentoid.util.Preferences;

import static androidx.core.view.ViewCompat.requireViewById;

/**
 * First-time use dialog for the choice of default app browse mode
 */
public class ReaderBrowseModeDialogFragment extends DialogFragment {

    public static void invoke(Fragment parent) {
        ReaderBrowseModeDialogFragment fragment = new ReaderBrowseModeDialogFragment();
        fragment.setCancelable(false);
        fragment.show(parent.getChildFragmentManager(), null);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedState) {
        return inflater.inflate(R.layout.dialog_reader_browse_mode_chooser, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View rootView, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(rootView, savedInstanceState);

        View ltrButton = requireViewById(rootView, R.id.chooseHorizontalLtr);
        ltrButton.setOnClickListener(v -> chooseBrowseMode(Preferences.Constant.VIEWER_BROWSE_LTR));

        View rtlButton = requireViewById(rootView, R.id.chooseHorizontalRtl);
        rtlButton.setOnClickListener(v -> chooseBrowseMode(Preferences.Constant.VIEWER_BROWSE_RTL));

        View verticalButton = requireViewById(rootView, R.id.chooseVertical);
        verticalButton.setOnClickListener(v -> chooseBrowseMode(Preferences.Constant.VIEWER_BROWSE_TTB));
    }

    private void chooseBrowseMode(int browseMode) {
        Preferences.setReaderBrowseMode(browseMode);
        getParent().onBrowseModeChange();
        dismiss();
    }

    private Parent getParent() {
        return (Parent) getParentFragment();
    }

    public interface Parent {
        void onBrowseModeChange();
    }
}

File: app/src/main/java/me/devsaki/hentoid/fragments/reader/ReaderBrowseModeDialogFragment.kt
package me.devsaki.hentoid.fragments.reader

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import me.devsaki.hentoid.databinding.DialogReaderBrowseModeChooserBinding
import me.devsaki.hentoid.fragments.BaseDialogFragment
import me.devsaki.hentoid.util.Settings
import me.devsaki.hentoid.util.Settings.Value.VIEWER_BROWSE_LTR
import me.devsaki.hentoid.util.Settings.Value.VIEWER_BROWSE_RTL
import me.devsaki.hentoid.util.Settings.Value.VIEWER_BROWSE_TTB

class ReaderBrowseModeDialogFragment : BaseDialogFragment<ReaderBrowseModeDialogFragment.Parent>() {
    companion object {
        fun invoke(parent: Fragment) {
            invoke(parent, ReaderBrowseModeDialogFragment())
        }
    }

    // UI
    private var binding: DialogReaderBrowseModeChooserBinding? = null


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedState: Bundle?
    ): View {
        binding = DialogReaderBrowseModeChooserBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onDestroyView() {
        binding = null
        super.onDestroyView()
    }

    override fun onViewCreated(rootView: View, savedInstanceState: Bundle?) {
        super.onViewCreated(rootView, savedInstanceState)

        binding?.apply {
            chooseHorizontalLtr.setOnClickListener {
                chooseBrowseMode(VIEWER_BROWSE_LTR)
            }
            chooseHorizontalRtl.setOnClickListener {
                chooseBrowseMode(VIEWER_BROWSE_RTL)
            }
            chooseVertical.setOnClickListener {
                chooseBrowseMode(VIEWER_BROWSE_TTB)
            }
        }
    }

    private fun chooseBrowseMode(browseMode: Int) {
        Settings.readerBrowseMode = browseMode
        parent?.onBrowseModeChange()
        dismiss()
    }

    interface Parent {
        fun onBrowseModeChange()
    }
}